### PR TITLE
8256048: Incomplete gitignore setting for netbeans project

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

The gitignore setting for netbeans project seems to be incomplete.
Only nbproject/private/ [1] is ignored. 

But there are also other files under nbproject/, which should be ignored too.
----------------------------
$ tree -L 1 nbproject/
nbproject/
├── configurations.xml
├── private
└── project.xml
----------------------------

It would be better to fix it.

Thanks.
Best regards,
Jie

[1] https://github.com/openjdk/jdk/blob/master/.gitignore#L5

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ❌ (1/2 failed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |    |  ✔️ (9/9 passed) |

**Failed test task**
- [Windows x64 (build release)](https://github.com/DamonFool/jdk/runs/1377040200)

### Issue
 * [JDK-8256048](https://bugs.openjdk.java.net/browse/JDK-8256048): Incomplete gitignore setting for netbeans project

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1122/head:pull/1122`
`$ git checkout pull/1122`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,7 +2,7 @@
 /dist/
 /.idea/
 /.vscode/
-nbproject/private/
+/nbproject/
```
Comment (erikj79) [inline .gitignore:5]: I believe the existing nbproject/private/ is meant to catch any private files created by someone opening any of the various nbproject directories that exist in the source base, so it should probably stay. Adding an ignore for any top level nbproject directory in addition to that should be ok.

Comment (DamonFool) [reply .gitignore]: Thanks @erikj79 for your review.
Updated.

Any comments?
Thanks.

Comment (erikj79) [reply .gitignore]: You misunderstand me. Adding the following line is fine and will make git ignore any nbproject dir in the root of the repository, which is what I assume you want:
/nbproject/

But removing the following existing line is not, as this matches any potential private part of already existing nbproject directories:
nbproject/private/

Comment (erikj79) [reply .gitignore]: Adding the below will make git ignore a bunch of already existing nbproject files.
nbproject/configurations.xml
nbproject/project.xml

Comment (DamonFool) [reply .gitignore:5]: Got it.
Thanks @erikj79 for your kind explanation.
Updated.

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**